src/WooCommerce.php

```diff
@@ -380,7 +385,7 @@ public static function getProductDetailsHtmlDataAttr(\WC_Product $product, $is_d
     $product_id = $product->get_id();
     $product_details = static::getProductDetails($product_id);
 
-    if ('variable' === $product->get_type() && $is_detail_view) {
+    if (('variable' === $product->get_type() && $is_detail_view) || (!isset($product_details['variant']) && 'variation' === $product->get_type())) {
```

Comment:
Keep the conditions consistent and test the major condition (product type) first.
```suggestion
    if (($product->get_type() === 'variable' && $is_detail_view) || ($product->get_type() === 'variation' && !isset($product_details['variant']))) {
```

Reply:
So the styling before I got here (Joan's styling) had the string first. I thought that's what you wanted me to stay consistent with. Of course this way is the way we are used to.